R/writeJsonResultsTo.R

```diff
@@ -73,6 +72,16 @@ writeJsonResultsToTable <- function(connectionDetails,
 
   DatabaseConnector::executeSql(connection = connection, sql = ddl, progressBar = TRUE)
 
+  # convert column names to snake case, omitting the autogenerated rowname column
+  # and the checkId column, which has no underscore in the results table DDL
+  for (i in 1:ncol(df)){
+    if (colnames(df)[i] %in% c("X_row","checkId")) {
+      colnames(df)[i] <- tolower(colnames(df)[i])
+    } else {
+      colnames(df)[i] <- SqlRender::camelCaseToSnakeCase(colnames(df)[i])
+    }
+  }
+
```

Comment:
A bit more color on this change.  Basically there is a column named `checkid` in the check results DDL queries ([example](https://github.com/OHDSI/DataQualityDashboard/blob/6c9c8fedb5a545dda6c34112acef40decda1fc5e/inst/sql/sql_server/result_table_ddl_field.sql#L21)) which should really be `check_id` to make for a consistent snakecase<>camelcase conversion.  However, I did not want to modify the DDL since I think that doing so could be considered a breaking change for some users who use the check results database table in downstream applications.

Regarding the `x_row` column - @clairblacketer do you know if this column serves any purpose in the code (searching the repo doesn't reveal any explicit references)?  I think it originates from a dataframe rowname column somewhere along the chain in the generation of the `checkResults` object (the X is pre-pended by R to make `_row` a valid colname).  If we apply camelcaseToSnakecase to this column, it will generate a different result than is in the DDL ("_x_row").  I was similarly hesitant to remove/change this col in the DDL in case that would be a breaking change for any users.

Both of the above mean we need this sorta clunky workaround to enable the camel to snakecase conversion.  I will definitely file an issue to clean this up in the next major release.

What do you think? 

Note there's a similar change on line 76 of `writeResultsTo.R`